core/src/main/java/com/graphhopper/routing/weighting/custom/EdgeToValueEntry.java

```diff
@@ -0,0 +1,32 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.routing.weighting.custom;
+
+import com.graphhopper.util.EdgeIteratorState;
+
+/**
+ * This internal interface calculates a double value from the specified EdgeIteratorState. Used for CustomWeighting.
+ *
+ * @see {{@link GeoToValueEntry}} or {@link EnumToValueEntry}
+ */
+interface EdgeToValueEntry {
```

Comment:
I am not fully convinced by this name yet. So we have edges and they have properties (=encoded values) and now we want a converter that takes an edge and returns a double value depending on one of these properties. Why `Entry`? To me `Entry` sounds like an element of a list or something and it seems very generic (not so easy to remember). Why (just) `Value` when we know it will be a (double) floating point number? 

Its a function that looks at a specific property of an edge and returns a number. Maybe even `EdgeDecoder` (it gives a meaning to a property `encoded` by the `EncodedValue`)? Or does this sound like there is only one correct way to 'decode' an encoded value (when really here the opposite is true and we have the freedom to specify how an 'encoded' value is turned into a number (as in `EncodedValueToDoubleConverter`). From the dictionary:

```
decode - to discover the meaning of information given in a secret or complicated way
```

We do not want to discover **the** meaning, but we rather want to **give** a meaning to the encoded information. So maybe more like an `EdgeInterpreter`? Or just `EdgeReader`? `(Boolean/Enum/Double)EncodedValueReader`?

Reply:
> I am not fully convinced by this name yet

It is indeed very similar to what the EncodedValues do and because it is related to the yaml/json structure I used "entry". Maybe EdgePropertyDecoder?

Feel free to change to something you like :) !

Reply:
`(Boolean/Enum/Double)EncodedValueDecoder` is kind of what it does except that it does not 'decode' **the** value, but rather assigns **a** value given the information of the encoded value. I think i would go with `EdgeDecoder` unless we have a better idea. In `EdgePropertyDecoder` I do not like `property` because its yet another word (calling the encoded values 'properties') and this makes it more confusing?

> it is related to the yaml/json structure

I don't think its necessarily only related to this. In fact I actually think these things are a very key concept: We import some graph (typically OSM) and store the OSM 'tags' along with the edges (in some efficient format, the encoded values). For routing we need a cost function, so we need something that interprets the 'encoded values' as numeric values and we can calculate edge weights. And this goes beyond the usage we have here for the new custom weighting stuff. For example I think `defaultSpeedMap` in `CarFlagEncoder` is also like an `EdgeToValueEntry`. So to me it seems this interface is not just an 'internal interface' but rather it will become a central part of GH and an important step for replacing FlagEncoder.

Reply:
`EncodedValueInterpreter`?

Reply:
> And this goes beyond the usage we have here for the new custom weighting stuff.

I'm not sure about this because in an arbitrary weighting you could directly "interpret" the value returned from `edge.get(yourEncodedValue)` 

> EncodedValueInterpreter?

ok :)

Reply:
Ok maybe I was thinking a bit too far ahead here. Lets stick with `EdgeToValueEntry` for now and think of them as the Java-side representation for the 'rules' defined in the yaml custom model. Also they are package private so right now they can be thought of as something specific for the custom model feature.